utility/serial.cpp

```diff
@@ -92,6 +92,71 @@ void ArdusatSerial::begin(unsigned long baud, bool setXbeeSpeed)
   }
 }
 
+/**
+ * Begin serial communications with a Sparkfun BlueSMiRF module at the specified baud
+ * rate.
+ *
+ * Note that baud rates above ~57600 are not well-supported by SoftwareSerial, and 
+ * even 57600 may cause some bugs.
+ *
+ * @param baud rate for serial communications
+ *
+ * NOTE: If this function is called multiple times with different values for 'baud'
+ *       without losing power in between calls, the baud rate will not be updated
+ *       after the first call.
+ */
+void ArdusatSerial::beginBluetooth(unsigned long baud)
+{
+  bool validBaud = true;
+  char baudCmd[8];
```

Comment:
Ah geeze - I forgot about the `'\0\'`. Thanks.

According to the bluetooth documentation the AT command to change the baud rate will never be more than 8 characters. The command to change the baudrate to 1200 is `U,1200,N`, and to change it to 115200 is `U,115K,N`. So `baudCmd[9]` should work for any baud rate.